web_sdk/web_test_utils/lib/skia_client.dart

```diff
@@ -0,0 +1,446 @@
+    }
+  }
+
+  /// Returns the latest positive digest for the given test known to Flutter
+  /// Gold at head.
+  Future<String?> getExpectationForTest(String testName) async {
+    late String? expectation;
+    final String traceID = getTraceID(testName);
+    await io.HttpOverrides.runWithHttpOverrides<Future<void>>(() async {
+      final Uri requestForExpectations = Uri.parse(
+        '$_skiaGoldHost/json/v2/latestpositivedigest/$traceID'
+      );
+      late String rawResponse;
+      try {
+        final io.HttpClientRequest request = await httpClient.getUrl(requestForExpectations);
+        final io.HttpClientResponse response = await request.close();
+        rawResponse = await utf8.decodeStream(response);
+        final dynamic jsonResponse = json.decode(rawResponse);
+        if (jsonResponse is! Map<String, dynamic>)
+          throw const FormatException('Skia gold expectations do not match expected format.');
+        expectation = jsonResponse['digest'] as String?;
+      } on FormatException catch (error) {
+        print(
+          'Formatting error detected requesting expectations from Flutter Gold.\n'
+          'error: $error\n'
+          'url: $requestForExpectations\n'
+          'response: $rawResponse'
+        );
+        rethrow;
+      }
+    },
+      SkiaGoldHttpOverrides(),
+    );
+    return expectation;
+  }
+
+  /// Returns a list of bytes representing the golden image retrieved from the
+  /// Flutter Gold dashboard.
+  ///
+  /// The provided image hash represents an expectation from Flutter Gold.
+  Future<List<int>>getImageBytes(String imageHash) async {
+    final List<int> imageBytes = <int>[];
+    await io.HttpOverrides.runWithHttpOverrides<Future<void>>(() async {
+      final Uri requestForImage = Uri.parse(
+        '$_skiaGoldHost/img/images/$imageHash.png',
+      );
+
+      try {
+        final io.HttpClientRequest request = await httpClient.getUrl(requestForImage);
+        final io.HttpClientResponse response = await request.close();
+        await response.forEach((List<int> bytes) => imageBytes.addAll(bytes));
+
+      } catch(e) {
+        rethrow;
+      }
+    },
+      SkiaGoldHttpOverrides(),
+    );
+    return imageBytes;
+  }
+
+  /// Returns the current commit hash of the Flutter repository.
+  Future<String> _getCurrentCommit() async {
+    final io.Directory webUiRoot = environment.webUiRootDir;
+    if (!webUiRoot.existsSync()) {
+      throw Exception('Web Engine root could not be found: $webUiRoot\n');
+    } else {
+      print('>> Web Engine Path: ${webUiRoot.path}');
+      final io.ProcessResult revParse = await process.run(
+        <String>['git', 'rev-parse', 'HEAD'],
+        workingDirectory: webUiRoot.path,
+      );
+      print('>>> rev parse:\n${revParse.stdout}\n<<<');
+      if (revParse.exitCode != 0) {
+        throw Exception('Current commit of Web Engine can not be found.');
+      }
+      return (revParse.stdout as String/*!*/).trim();
```

Comment:
What's the `/*!*/` for? (applies here and elsewhere in this file)

Reply:
Copy-pasta from the framework. Removed.